provider/spark.go

```diff
@@ -37,30 +37,45 @@ type SparkExecutorType string
 
 const (
 	EMR        SparkExecutorType = "EMR"
-	Databricks                   = "DATABRICKS"
+	Databricks SparkExecutorType = "DATABRICKS"
 )
 
 type JobType string
 
 const (
 	Materialize       JobType = "Materialization"
-	Transform                 = "Transformation"
-	CreateTrainingSet         = "Training Set"
+	Transform         JobType = "Transformation"
+	CreateTrainingSet JobType = "Training Set"
 )
 
 const MATERIALIZATION_ID_SEGMENTS = 3
 const ENTITY_INDEX = 0
 const VALUE_INDEX = 1
 const TIMESTAMP_INDEX = 2
 
-type SparkExecutorConfig []byte
+type AWSCredentials struct {
+	AWSAccessKeyId string
+	AWSSecretKey   string
+}
+
+type SparkExecutorConfig interface {
+	Serialize() []byte
+	Deserialize(config SerializedConfig) error
+	IsExecutorConfig() bool
+}
+
+type SparkFileStoreConfig interface {
+	Serialize() []byte
+	Deserialize(config SerializedConfig) error
+	IsFileStoreConfig() bool
+}
 
 type SparkConfig struct {
 	ExecutorType   SparkExecutorType
-	ExecutorConfig DatabricksConfig
+	ExecutorConfig SparkExecutorConfig
 	StoreType      FileStoreType
-	StoreConfig    AzureFileStoreConfig
-} //TODO, change these back to type agnostic after databricks tests
+	StoreConfig    SparkFileStoreConfig
```

Comment:
Does this need to be SparkFileStoreConfig or can it just be FilestoreConfig?

Reply:
FileStoreConfig already exists; it's of type `[]byte`